src/components/ViewSnippetHeader.tsx

```diff
@@ -62,6 +62,8 @@ export default function ViewSnippetHeader() {
     },
   })
 
+  const [isStarred, setIsStarred] = useState(false)
```

Comment:
shouldn't you use `snippet.is_starred` here?

Reply:
yeah, but it'll trigger the unstar method, and will eventually unstar the snippet even before the user wants it to.

ig, I have to remove this hook for now.

Reply:
`/snippets/get` returns `snippet.is_starred`. Not sure what you mean by it'll trigger the unstar method

Reply:
If `/snippets/get` doesn't return `snippet.is_starred` with the specific property set for the user, then that has to be implemented as well

Reply:
ohh, okay. so it means i have to `/snippets/get` too. and use it inside the isStarred hook.

i did it earlier, but u asked to remove that method

Reply:
idk what the isStarred hook is. You don't need to make any new hooks afaik

when you run `qc.invalidateQueries(["snippets", snippet!.snippet_id])` it automatically causes the snippet to be reloaded

Reply:
yeah i've removed it now. it was from one of my previous commits, and i forgot to do it in the past commit.

Reply:
@seveibar can u look into this comment and just update me

Reply:
yes you don't need `useState` or `isStarred`, in fact it is bad to have it because it confuses the source of truth of whether or not the snippet is starred